app/controllers/missions_controller.rb

```diff
@@ -4,7 +4,8 @@ class MissionsController < ApplicationController
   before_action :find_mission, only: %i[edit update destroy]
 
   def index
-    if valid_order?(params[:order])
+    @key = Mission.ransack(params[:search_key])
```

Comment:
Nit: 
也許可以使用更常見的 [`@q`](https://github.com/activerecord-hackery/ransack#in-your-controller) or `@query` 當作 query 條件的變數名稱：
```suggestion
    @q = Mission.ransack(params[:q])
```